src/Conversion/ONNXToTOSA/Math/Softmax.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- Elementwise.cpp - Elementwise Op --------------------===//
+//
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file lowers ONNX element-wise operators to TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/Tosa/IR/TosaOps.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/TypeUtilities.h"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.hpp"
+#include "src/Dialect/ONNX/ONNXOps.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+class ONNXSoftmaxLoweringToTOSA : public OpConversionPattern<ONNXSoftmaxOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  using OpAdaptor = typename ONNXSoftmaxOp::Adaptor;
+  LogicalResult matchAndRewrite(ONNXSoftmaxOp op, OpAdaptor adaptor,
```

Comment:
What happens when the `axis` attribute is set? Is it not supported by TOSA?

Reply:
I missed that. Added this in [550cc52](https://github.com/ljfitz/onnx-mlir/pull/65/commits/550cc52a6963ff85ac934bbc3bd674ee795f0e38)